src/main/java/com/northconcepts/datapipeline/foundations/examples/pipeline/FilterColumnsWithAllNullValues.java

```diff
@@ -14,24 +23,37 @@ public static void main(String[] args) {
 
         LocalFile inputFile = new LocalFile()
                 .setName("Input File")
-                .setPath("PATH_TO_INPUT_FILE.csv")
+                .setPath("data/input/Listing.csv")
                 .detectFileType();
 
-        ExcelPipelineInput pipelineInput = new ExcelPipelineInput()
+        LocalFile outputFile = new LocalFile()
+                .setName("Output File")
+                .setPath("data/output/NoNull.xlsx");
+
+        CsvPipelineInput pipelineInput = new CsvPipelineInput()
                 .setFileSource(inputFile)
                 .setFieldNamesInFirstRow(true);
 
+        ExcelPipelineOutput pipelineOutput = new ExcelPipelineOutput()
+                .setFileSink(outputFile)
+                .setFieldNamesInFirstRow(true);
+
         pipeline.setInput(pipelineInput);
+        pipeline.setOutput(pipelineOutput);
 
         Dataset dataset = pipeline.getDataset();
         dataset.load().waitForRecordsToLoad();
 
+        SelectArrangeFieldsCommand action = new SelectArrangeFieldsCommand();
+
         for(Column column : dataset.getColumns()) {
-            if(column.getValueCount() == column.getNullCount()) {
-                continue;
+            if(column.getValueCount() != column.getNullCount()) {
+                action.add(column.getName());
             }
-            System.out.println(column.getName());
         }
 
+        pipeline.addAction(action);
+
+        Job.run(pipeline.createDataReader(true), pipeline.createDataWriter());
```

Comment:
Can we use `dataset.load().waitForRecordsToLoad();` instead of working directly with the data readers and writers?

Reply:
@DeleTaylor Won't the DataWriter used in dataset.load() only write to MVStore? Which would mean that there would be no output file produced? You requested to write to an output file in your first comment for this class.

Reply:
@Bryan-Noel excellent catch.  Thanks for looking into the details of my requests.

I've added the ability to create and run jobs directly from pipelines (`Pipeline.run()` in this case).  Can you try using the new method in these examples?